widget/editor.go

```diff
@@ -936,6 +1171,153 @@ func (e *Editor) NumLines() int {
 	return len(e.lines)
 }
 
+func (e *Editor) IsSelected() bool {
+	return e.selection.len() > 0
+}
+
+// Selection returns the start and end of the selection, as offsets into the
+// editor text. start can be > end.
+func (e *Editor) Selection() (start, end int) {
+	return e.selection.start.ofs, e.selection.end.ofs
+}
+
+// SetSelection sets the selection start and end, which are offsets into the
+// editor text.
+func (e *Editor) SetSelection(start, end int) {
+	e.selection.start.ofs, e.selection.end.ofs = start, end
+	e.selection.makeValid(e)
+}
+
+// SelectedText returns the currently selected text (if any) from the editor.
+func (e *Editor) SelectedText() string {
+	start, end := sortInts(e.selection.start.ofs, e.selection.end.ofs)
+
+	// Do I really need these?
```

Comment:
Well do ya'? :)

Reply:
😆 I suspect I don't any more, now that those fields are not exported any more.